pkg/bucket/replication/replication.go

```diff
@@ -187,9 +199,11 @@ func (c Config) Replicate(obj ObjectOpts) bool {
 			default:
 				return rule.DeleteMarkerReplication.Status == Enabled
 			}
-		} else { // regular object/metadata replication
+		} // regular object/metadata replication
+		if !obj.Replica {
 			return true
 		}
+		return obj.Replica && rule.SourceSelectionCriteria.ReplicaModifications.Status == Enabled
```

Comment:
```suggestion
		return obj.Replica && rule.SourceSelectionCriteria.ReplicaModifications.Status == Enabled
```

Can this moved as a function for `rule.MetadataReplicate()` ?